File: back-end/community/Observer.py
from typing import List
from ..contracts.Contracts import my_token_contract


class Observer():
    def get_all(self, account: str) -> List[Achievement]:
        return my_token_contract.functions.getAllAchievements(account).call()

    def get_open(self, account: str) -> List[Achievement]:
        return my_token_contract.functions.getOpenAchievements(account).call()

File: back-end/js/community/Observer.js
class Observer {
    async get_all_achievements() {
        let account = await getCookie('account');
        let all_achievements = [];
        all_achievements = await my_token_contract.methods.getAllAchievements(account).call()
        return all_achievements
    }

    async get_open_achievements() {
        let account = await getCookie('account');
        let open_achievements = []
        open_achievements = await my_token_contract.methods.getAllAchievements(account).call()
        open_achievements.filter(item => item.is_private)
        return open_achievements
    }
}
